corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -397,45 +398,98 @@ def get_case_payload_batched(domain, stock_settings, version, user, last_synclog
-                            If None, default settings will be used.
-    :param domain:          The domain object. An instance of `Domain`.
     :param force_cache:     Set to `True` to force the response to be cached.
-                            Only applies if `restore_id` is empty.
     :param cache_timeout:   Override the default cache timeout of 1 hour.
-                            Only applies if `restore_id` is empty.
     :param overwrite_cache: Ignore any previously cached value and re-generate the restore response.
-                            Only applies if `restore_id` is empty.
     """
-    def __init__(self, user, restore_id="", version=V1, state_hash="",
-                 items=False, stock_settings=None, domain=None, force_cache=False,
-                 cache_timeout=None, overwrite_cache=False):
+
+    def __init__(self, force_cache=False, cache_timeout=None, overwrite_cache=False):
+        self.force_cache = force_cache
+        self.cache_timeout = cache_timeout if cache_timeout is not None else INITIAL_SYNC_CACHE_TIMEOUT
+        self.overwrite_cache = overwrite_cache
+
+class RestoreState(object):
+
+    def __init__(self, user, params):
         self.user = user
-        self.restore_id = restore_id
-        self.version = version
-        self.state_hash = state_hash
-        self.items = items
+        self.params = params
 
-        if stock_settings:
-            self.stock_settings = stock_settings
-        elif domain and domain.commtrack_settings:
+    @property
+    @memoized
+    def sync_log(self):
+        if self.params.sync_log_id:
+            try:
+                sync_log = SyncLog.get(self.params.sync_log_id)
+            except ResourceNotFound:
+                # if we are in loose mode, return an HTTP 412 so that the phone will
+                # just force a fresh sync
+                raise MissingSyncLog('No sync log with ID {} found'.format(self.params.sync_log_id))
+            if sync_log.doc_type != 'SyncLog':
+                raise InvalidSyncLogException('Bad sync log doc type for {}'.format(self.params.sync_log_id))
+            elif sync_log.user_id != self.user.user_id:
+                raise SyncLogUserMismatch('Sync log {} does not match user id {} (was {})'.format(
+                    self.params.sync_log_id, self.user.user_id, sync_log.user_id
+                ))
+
+            return sync_log
+        else:
+            return None
+
+
+class RestoreConfig(object):
+    """
+    A collection of attributes associated with an OTA restore
+
+    :param domain:          The domain object. An instance of `Domain`.
+    :param user:            The mobile user requesting the restore
+    :param params:          The RestoreParams associated with this (see above).
+    :param cache_settings:  The RestoreCacheSettings associated with this (see above).
+    """
+
+    def __init__(self, domain=None, user=None, params=None, cache_settings=None):
```

Comment:
i was pretty deliberate in how i tested/changed this signature. you can look at the commit history if you're curious.